### PR TITLE
balancergroup: improve observability around balancer cache behavior

### PR DESCRIPTION
**Pull request by easwars**
RELEASE NOTES: none

### CODE REVIEW

**Review by dfawley — COMMENTED** [review]
Should these be at a higher verbosity?  I don't think we want to log these by default for everyone, do we?

**Comment by easwars** [comment]
> Should these be at a higher verbosity? I don't think we want to log these by default for everyone, do we?

Fair point. Moved them under `V2` and also fixed one outlier log which had a `balancer group` prefix. We are already using a prefix logger here.

File: internal/balancergroup/balancergroup.go [hunk 1]
```diff
@@ -403,7 +408,7 @@ func (bg *BalancerGroup) Remove(id string) {
 
 	sbToRemove, ok := bg.idToBalancerConfig[id]
 	if !ok {
-		bg.logger.Infof("balancer group: trying to remove a non-existing locality from balancer group: %v", id)
+		bg.logger.Warningf("Child policy for locality %q does not exist in the balancer group", id)
```
Comment (dfawley) [inline internal/balancergroup/balancergroup.go:411]: Should this even be `Error` so our tests fail?  I.e. does it imply a coding error on our part?

**Review by dfawley — APPROVED** [review]
Approving but please see the inline question.

**Pull request merged**